### ISSUE TITLE
Ensure Consistent Ordering of Organizations by the Creation Time in Email Domain Discovery

### ISSUE THREAD

**Issue by BimsaraBodaragama** [wso2/product-is#20998]
**Is your suggestion related to an experience? Please describe.**
In the Email Domain Discovery section under Login & Registration, the list of organizations does not follow the same order by creation time as other endpoints, such as the GET _/organizations_ endpoint, which returns the organizations with the most recently added at the top. This inconsistency in ordering can lead to confusion for users, as they might expect the same behavior across the platform.

**Describe the improvement**
The ordering of organizations in the Email Domain Discovery list should be modified to align with the behavior of other endpoints. Specifically, the list should be ordered by the creation time of the email domain items, ensuring that the most recently added organization appears at the top.

**Additional context**
Currently, the sorting in the Email Domain Discovery section is done based on the organization ID, rather than by creation time. This leads to a randomized order as more items are added. Aligning the ordering logic with other endpoints will provide a consistent user experience.

<img width="1800" alt="Email-Domain-Discovery-GET-Improvement-Issue" src="https://github.com/user-attachments/assets/ddab0bcc-9228-4603-b365-88948fae8dbd">

**Comment by SandaruRF** [comment]
Hi @BimsaraBodaragama , I'd like to work on this issue. Could you please assign it to me?

**Comment by BimsaraBodaragama** [comment]
> Hi [@BimsaraBodaragama](https://github.com/BimsaraBodaragama) , I'd like to work on this issue. Could you please assign it to me?

Hi @SandaruRF,
Sure. I’ve assigned you the issue. You can start on the fix.

**Comment by SandaruRF** [comment]
> Hi [@SandaruRF](https://github.com/SandaruRF), Sure. I’ve assigned you the issue. You can start on the fix.

Thank you

**Comment by sagara-gunathunga** [comment]
@SandaruRF Just checking in to see if you’re still working on this issue. If you are, it would be great to share a quick update or a draft PR to show your progress. If you’re no longer working on it, please let us know so we can make the issue available to other participants.

**Comment by SandaruRF** [comment]
> [@SandaruRF](https://github.com/SandaruRF) Just checking in to see if you’re still working on this issue. If you are, it would be great to share a quick update or a draft PR to show your progress. If you’re no longer working on it, please let us know so we can make the issue available to other participants.

Yes, I’m still working on this. I’ll share a draft PR with the progress soon.

**Comment by pavinduLakshan** [comment]
> > [@SandaruRF](https://github.com/SandaruRF) Just checking in to see if you’re still working on this issue. If you are, it would be great to share a quick update or a draft PR to show your progress. If you’re no longer working on it, please let us know so we can make the issue available to other participants.
> 
> Yes, I’m still working on this. I’ll share a draft PR with the progress soon.

Hi @SandaruRF, were you able to open a pull request yet?

**Comment by pavinduLakshan** [comment]
This issue has been unassigned due to prolonged inactivity. It is now available for other contributors to pick up.

**Comment by Yasodhyassp30** [comment]
Hi I would like to work on this issue can i get this reserved ?

**Comment by BimsaraBodaragama** [comment]
> Hi I would like to work on this issue can i get this reserved ?

Hi @Yasodhyassp30,
Sure. I’ve assigned you the issue. You can start on the fix.

**Comment by BimsaraBodaragama** [comment]
Hi @Yasodhyassp30,

Could you please provide an update on this issue? 
Feel free to attach a draft PR if you are in the middle of the fix.

Thanks!
BB

**Comment by Yasodhyassp30** [comment]
Hi,

I was not able to start on this due to health issues, I'm currently hospitalised as well, I think I won't be able to complete within the given time
Really sorry for the inconvenience 

Thank you

**Comment by sagara-gunathunga** [comment]
@Yasodhyassp30 sorry to hear that and thanks for updating us. Hope you feel better soon.

**Comment by ApiramR** [comment]
Hi @BimsaraBodaragama , I'd like to work on this issue. Could you please assign it to me?

**Comment by BimsaraBodaragama** [comment]
> Hi [@BimsaraBodaragama](https://github.com/BimsaraBodaragama) , I'd like to work on this issue. Could you please assign it to me?

Hi @ApiramR,
Sure. I’ve assigned you the issue. You can start on the fix.

**Comment by ApiramR** [comment]
@BimsaraBodaragama I have added the PR for this issue. Could you please review it. Thanks.

**Comment by BimsaraBodaragama** [comment]
> [@BimsaraBodaragama](https://github.com/BimsaraBodaragama) I have added the PR for this issue. Could you please review it. Thanks.

@ApiramR,

Please link the PR in the issue.

Thanks!
BB

**Comment by ApiramR** [comment]
Hi @BimsaraBodaragama,

Here is the PR for this issue: https://github.com/wso2-extensions/identity-organization-management/pull/597

Thanks!

**Comment by ApiramR** [comment]
Hi @BimsaraBodaragama 
Any Updates on the PR review?

**Comment by BimsaraBodaragama** [comment]
> Hi [@BimsaraBodaragama](https://github.com/BimsaraBodaragama) Any Updates on the PR review?

Hi @ApiramR,

The PR is being reviewed by the team. We will give you an update.

Thanks,
BB

**Comment by BimsaraBodaragama** [comment]
> Hi [@BimsaraBodaragama](https://github.com/BimsaraBodaragama) Any Updates on the PR review?

Hi @ApiramR,

We have added some comments.

Thanks,
BB

**Comment by ApiramR** [comment]
Hi @BimsaraBodaragama  Thanks for the comments, I made changes according to that. Could you please check it. Thank you.

**Comment by ApiramR** [comment]
Hii @BimsaraBodaragama Any updates on the review?

**Comment by BimsaraBodaragama** [comment]
> Hii [@BimsaraBodaragama](https://github.com/BimsaraBodaragama) Any updates on the review?

cc: @SujanSanjula96

**Comment by ApiramR** [comment]
Hi, @BimsaraBodaragama @SujanSanjula96 Just following up on this PR. Please let me know if there’s anything else needed from my side. Thank you.

**Comment by BimsaraBodaragama** [comment]
> Hi, [@BimsaraBodaragama](https://github.com/BimsaraBodaragama) [@SujanSanjula96](https://github.com/SujanSanjula96) Just following up on this PR. Please let me know if there’s anything else needed from my side. Thank you.

Hi @ApiramR,

We’ve reviewed the changes and added a few comments. Please address them so we can proceed with the merge.

Thanks,
BB

**Comment by ApiramR** [comment]
Thanks @BimsaraBodaragama I updated according to that.

**Comment by SujanSanjula96** [comment]
Hi @ApiramR,

We have merged the PR [1].  Hence, we are closing the issue. Thanks for the contribution.

[1] - https://github.com/wso2-extensions/identity-organization-management/pull/597

**Comment by BimsaraBodaragama** [comment]
You can track the PR here
 - https://github.com/wso2-extensions/identity-organization-management/pull/597
